components/motor/gpio/basic.go

```diff
@@ -186,39 +183,51 @@ func (m *Motor) setPWM(ctx context.Context, powerPct float64, extra map[string]i
 		errs = multierr.Combine(errs, m.EnablePinHigh.Set(ctx, true, extra))
 	}
 
+	m.powerPct = powerPct
+	powerPct = math.Min(powerPct, m.maxPowerPct)
+	powerPct = math.Max(powerPct, -1*m.maxPowerPct)
+
+	// pin calls on builtin baords will only take poisitve power, so the absolute value is required to be passed
+	// to pwm pins.
+	pwmPower := math.Abs(powerPct)
```

Comment:
kk so it turns out if we do an abs here, the power isn't negative anymore so the switch statement doesn't work

Reply:
I'm fine with that, I want to add this to rpmMonitor in the next refactor so I agree w/ unblocking returning negative.